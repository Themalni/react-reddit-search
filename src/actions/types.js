export const FETCH_POSTS = 'FETCH_POSTS';
export const SEARCH_POSTS = 'SEARCH_POSTS';
export const IS_LOADING_CHANGE_STATE = 'IS_LOADING_CHANGE_STATE';
export const POSTS_FOUND_CHANGE_STATE = 'FOUND_POSTS_CHANGE_STATE';
export const POSTS_NOT_FOUND_CHANGE_STATE = 'POSTS_NOT_FOUND_CHANGE_STATE';