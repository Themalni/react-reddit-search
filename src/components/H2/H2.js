import styled from 'styled-components'

const H2 = styled.h2`
  align-self: center;
  font-size:  1.2em;
  color:  ${(props) => props.theme.colors.white}
  font-weight: ${(props) => props.theme.font.bold}
`

export default H2