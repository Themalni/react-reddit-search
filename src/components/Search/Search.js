import React, { Component } from 'react';
import { Container } from '../Container/Container';
import Form from '../Form/Form';
import { InputText } from '../Input/Input';
import { Button } from '../Button/Button';
import { connect } from 'react-redux';
import { searchPosts } from '../../actions/searchActions';
import { fetchPosts, isLoadingStateChange, postsNotFoundChangeState } from '../../actions/postsActions';
import PropTypes from 'prop-types';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: ''
    }
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
    this.props.postsNotFoundChangeState(false);
  }

  onSubmit = (e) => {
    e.preventDefault();

    const searchTerm = {
      search: this.state.search
    }
    this.props.searchPosts(searchTerm)
    this.props.fetchPosts()
    this.props.isLoadingStateChange(true)
  }

  render() {
    return (
      <Container>
        <Form onSubmit={this.onSubmit}>
          <InputText type="text" placeholder="Enter search term..." name="search" onChange={this.onChange} value={this.state.search} required />
          <Button>Search</Button>
        </Form>
      </Container>
    )
  }
}

Search.propTypes = {
  searchPosts: PropTypes.func.isRequired
}

const mapStateToProps = (state) => {
  return {
    searchTerm: state.search.searchTerm,
    posts: state.posts.items,
    postsNotFound: state.posts.postsNotFound
  }
}

export default connect(mapStateToProps, { searchPosts, fetchPosts, isLoadingStateChange, postsNotFoundChangeState })(Search)