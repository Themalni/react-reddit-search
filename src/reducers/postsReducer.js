import { FETCH_POSTS, POSTS_FOUND_CHANGE_STATE, IS_LOADING_CHANGE_STATE, POSTS_NOT_FOUND_CHANGE_STATE } from '../actions/types';

export const initState = {
  postsFound: false,
  postsNotFound: false,
  isLoading: false,
  items: []
}

export function postsReducer(state = initState, action) {
  switch (action.type) {
    case FETCH_POSTS:
      return {
        ...state,
        items: action.payload
      }
    case IS_LOADING_CHANGE_STATE:
      return {
        ...state,
        isLoading: action.payload
      }
    case POSTS_FOUND_CHANGE_STATE:
      return {
        ...state,
        postsFound: action.payload
      }
    case POSTS_NOT_FOUND_CHANGE_STATE:
      return {
        ...state,
        postsNotFound: action.payload
      }
    default:
      return state;
  }
}