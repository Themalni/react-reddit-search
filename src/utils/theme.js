import { colors } from './colors';

export const theme = {
  colors,
  font: {
    thin: 300,
    regular: 400,
    bold: 700,
  },
  media: {
    phone: '@media (min-width: 320px)',
    tablet: '@media (min-width: 728px)',
    laptop: '@media (min-width: 1024px)',
    desktop: '@media (min-width: 1280px)'
  }
}