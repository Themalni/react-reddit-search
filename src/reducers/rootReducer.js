import { combineReducers } from 'redux';
import { postsReducer } from './postsReducer';
import searchReducer from './searchReducer';

export default combineReducers({
  posts: postsReducer,
  search: searchReducer
})