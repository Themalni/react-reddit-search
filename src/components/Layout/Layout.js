import React from 'react';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import { theme } from '../../utils/theme';

const GlobalStyle = createGlobalStyle`

@import url('https://fonts.googleapis.com/css?family=Nunito:300,400,700&display=swap');

  body {
    padding: 0;
    margin: 0;
    font-family: 'Roboto';
    color: white;
    font-family: 'Nunito', sans-serif;
    background: ${({ theme }) => theme.colors.grey};
  }

  *, *::before, *::after {
    box-sizing: border-box;
  }
`;

const Wrapper = styled.div`
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
`;

const Layout = ({ children }) => (
  <ThemeProvider theme={theme}>
    <>
      <GlobalStyle />
      <Wrapper>
        {children}
      </Wrapper>
    </>
  </ThemeProvider>
);

export default Layout;