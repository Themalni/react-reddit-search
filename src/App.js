import React, { Component } from 'react';
import Layout from './components/Layout/Layout';
import H1 from './components/H1/H1';
import Search from './components/Search/Search';
import Post from './components/Post/Post';
import { connect } from 'react-redux';
import Loader from './components/Loader/Loader'
import { LowerContainer } from './components/Container/Container';
import H2 from './components/H2/H2'

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
    };
  }
  render() {

    return (
      <Layout>
        <H1 isLarge>Search Reddit app</H1>
        <Search></Search>
        {this.props.postsNotFound ? <H2>Sorry, nothing found for "{this.props.searchTerm}"</H2> : null}
        {this.props.isLoading ? <LowerContainer><Loader /></LowerContainer> : null}
        {this.props.postsFound ? <Post /> : null}
      </Layout >
    );
  }
}

const mapStateToProps = (state) => {
  console.log("SearchTerm:", state.search.searchTerm)
  return {
    postsFound: state.posts.postsFound,
    postsNotFound: state.posts.postsNotFound,
    isLoading: state.posts.isLoading,
    searchTerm: state.search.searchTerm.search
  }
}

export default connect(mapStateToProps)(App)