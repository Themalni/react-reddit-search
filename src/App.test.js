import React from 'react';
import { createStore } from 'redux';
import { render } from '@testing-library/react'
import { Provider } from 'react-redux';
import rootReducer from './reducers/rootReducer';
import { initState } from './reducers/postsReducer';
import { FETCH_POSTS, POSTS_FOUND_CHANGE_STATE, IS_LOADING_CHANGE_STATE, POSTS_NOT_FOUND_CHANGE_STATE } from './actions/types';
import App from './App';
import Layout from './components/Layout/Layout';
import H2 from './components/H2/H2';
import Search from './components/Search/Search';
import { theme } from './utils/theme';
import { ThemeProvider } from 'styled-components';


function renderWithRedux(component,
  { initState, store = createStore(rootReducer, initState) } = {}) {
  return {
    ...render(<Provider store={store}><ThemeProvider theme={theme}>{component}</ThemeProvider></Provider>),
    store
  }
}

describe('App', () => {
  it('renders correctly', () => {
    const { container } = renderWithRedux(<App />);
    expect(container.firstChild).toMatchSnapshot();
  })
})