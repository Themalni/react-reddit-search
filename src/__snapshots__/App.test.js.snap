// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  class="sc-bdVaJa EGqKq"
>
  <h1
    class="sc-bwzfXH hpwBDX"
  >
    Search Reddit app
  </h1>
  <div
    class="sc-htpNat kmuuwR"
  >
    <form
      class="sc-EHOje enRHgQ"
    >
      <input
        class="sc-bZQynM bUyRsv"
        name="search"
        placeholder="Enter search term..."
        required=""
        type="text"
        value=""
      />
      <button
        class="sc-gzVnrw cQXIoY"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;
